tests/bluetooth/df/connectionless_cte_chains/src/test_remove_cte_from_chain.c

```diff
@@ -149,17 +146,199 @@ void test_remove_cte_from_single_pdu_chain(void)
+
+	/* Free PDUs fifo should hold single PDU released during double buffer swap. The PDU
+	 * was allocated during advertising set creation.
+	 */
+	pdu_fifo_cnt = lll_adv_free_pdu_fifo_count_get();
+	zassert_equal(pdu_fifo_cnt, TEST_PER_ADV_SINGLE_PDU,
+		      "Unexpected number of free PDUs in a fifo: %d", pdu_fifo_cnt);
+
+	/* Expected free PDUs count is decreased by:
+	 * - single PDU allocated during advertising set creation,
+	 * - number of PDUs allocated for per. adv. chain to Tx CTE
+	 */
+	pdu_mem_cnt_expected -= expected_mem_pdu_used_count_for_enable;
+	pdu_mem_cnt = lll_adv_pdu_mem_free_count_get();
+	zassert_equal(pdu_mem_cnt, pdu_mem_cnt_expected,
+		      "Unexpected amount of free PDUs memory: %d, expected %d", pdu_mem_cnt,
+		      pdu_mem_cnt_expected);
+
+	if (new_chain_before_cte_disable) {
+		common_create_per_adv_chain(adv, updated_chain_length);
+	}
+
+	err = ll_df_set_cl_cte_tx_enable(handle, false);
+	zassert_equal(err, 0,
+		      "Unexpected error while disabling CTE for periodic avertising chain, err: %d",
+		      err);
+	/* Validate result */
+	common_validate_per_adv_chain(adv, expected_pdu_in_chain_after_cte_disable);
+
+	/* Swap PDU double buffer to check correctness of release former PDUs */
+	pdu_new = lll_adv_sync_data_latest_get(adv->lll.sync, NULL, &upd);
+	zassert_not_equal(pdu_new, NULL,
+			  "Unexpected value of PDU pointer after PDU double buffer swap");
+
+	/* Validate number of released PDUs */
+
+	/* Number of free PDUs in a fifo is a number of released PDUs from former periodic
+	 * advertising chain. One free PDU that had been in a fifo was used for allocation of
+	 * the new PDU without CTE.
+	 */
+	pdu_fifo_cnt = lll_adv_free_pdu_fifo_count_get();
+	zassert_equal(pdu_fifo_cnt, expected_end_fifo_free_pdu_count,
+		      "Unexpected number of free PDUs in a fifo: %d", pdu_fifo_cnt);
+
+	/* Number of free PDUs in memory pool may decreased. Single PDU for AUX_SYNC_IND was
+	 * acquired from free PDUs fifo. The memory pool will decrease by number of non empty PDUs
+	 * in a chain subtracted by 1 (PDU taken from free PDUs fifo).
+	 */
+	pdu_mem_cnt_expected -= expected_mem_pdu_used_count_for_disable;
+
+	pdu_mem_cnt = lll_adv_pdu_mem_free_count_get();
+	zassert_equal(pdu_mem_cnt, pdu_mem_cnt_expected,
+		      "Unexpected amount of free PDUs memory: %d, expected %d", pdu_mem_cnt,
+		      pdu_mem_cnt_expected);
+
+	common_teardown(adv);
+}
+
+void test_remove_cte_from_chain_extended_to_tx_all_cte_after_enqueue_to_lll(void)
+{
+	uint8_t cte_count = TEST_CTE_COUNT;
+	uint8_t init_chain_length = TEST_PER_ADV_SINGLE_PDU;
+	uint8_t expected_mem_pdu_used_count_for_enable = TEST_CTE_COUNT + TEST_PER_ADV_SINGLE_PDU;
+	uint8_t expected_mem_pdu_used_count_for_disable = 0;
+	uint8_t expected_pdu_in_chain_after_cte_disable = TEST_PER_ADV_SINGLE_PDU;
+	uint8_t updated_chain_length = 0;
+	uint8_t expected_end_fifo_free_pdu_count = TEST_CTE_COUNT;
+	bool new_chain_before_cte_disable = false;
```

Comment:
Not that it really matters for a unit test, but these could have been `const`

Reply:
Yean and written directly in function call. I did it just to make values to arguments assignment easier to read.

Reply:
I like the named variables :) It makes it much easier to understand, especially with this amount of arguments.